src/components/PendingSwapExchange.tsx

```diff
@@ -131,7 +136,8 @@ const PendingSwapExchange = ({
       >
         {t("settleAsToken")}
         <img
```

Comment:
Wow! This whole PR is beautiful. Would it be possible to create a new component like `TokenIcon` that just takes `symbol` as a prop (and arbitrary `img` element props) and handles the `getTokenIconPath` and `onError` behind the scenes?

Reply:
Got you! 👍